File: pset6/sentimental/vigenere/vigenere.py
import sys

def main():
    if (len(sys.argv) != 2):
        # print("input 2 command line arguments")
        exit(1)

    alpha = True
    for k in range(len(sys.argv[1])):
        if not sys.argv[1].isalpha():
            alpha = False;

    if (not alpha):
        # print("key is not alphabetical")
        exit(1)

    text = input("plaintext: ");

    list = cipher(text, sys.argv[1])

    print("ciphertext: ", end="")

    for a in range(len(list)):
        print(list[a], end="")

    print()

    exit(0)

def cipher(text, key):
    list = []
    k = 0

    for p in range(len(text)):

        if (k + 1 > len(key)):
            k = 0

        c = 0

        if (text[p].isalpha()):
            if text[p].isupper():
                c = (((ord(text[p])) - 65) + (ord(key[k].upper()) - 65)) % 26
                c += 65
                list.append(chr(c))
            else:
                c = (((ord(text[p])) - 97) + (ord(key[k].upper()) - 65)) % 26
                c += 97
                list.append(chr(c))

        else:
            list.append(text[p])
            k -= 1

        k += 1

    return list

if __name__ == "__main__":
    main()

File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int cipher(string plainText, string key);

int main(int argc, string argv[])           // Command line: gets key
{
    string key = argv[1];

    if (argc != 2)                  // Checks if key is invalid
    {
        printf("Keyword is invalidddddd\n");

        return 1;
    }
    else
    {
        for (int k = 0, n = strlen(key); k < n; k++)
        {
            if (!isalpha(key[k]))
            {
                printf("Keyword is invaliddddddddddd\n");

                return 1;
            }
        }
    }

    string plainText = get_string("plaintext: ");         // gets plaintext

    printf("ciphertext: ");
    cipher(plainText, key);
    printf("\n");

    return 0;
}

int cipher(string plainText, string key)         // turns plaintext into ciphertext
{
    for (int p = 0, k = 0, n = strlen(plainText); p < n; p++, k++)
    {
        if (k + 1 > strlen(key))          // loops key if plaintext is longer
        {
            k = 0;
        }

        int c = 0;

        if (isalpha(plainText[p]))          // prints alphabetic characters
        {
            if (isupper(plainText[p]))      // sets uppercase characters to proper numbers and converts to ciphertext
            {
                c = ((plainText[p] - 65) + (toupper(key[k]) - 65)) % 26;
                c = c + 65;
                printf("%c", (char)c);
            }
            else                            // sets lowercase '' and converts back to lowercase
            {
                c = ((plainText[p] - 97) + (toupper(key[k]) - 65)) % 26;
                c = c + 97;
                printf("%c", (char)c);
            }
        }
        else                // prints non-alphabetic characters
        {
            printf("%c", plainText[p]);
            k--;
        }
    }

    return 0;
}